packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/knowledge/KnowledgeController.java

```diff
@@ -129,6 +139,10 @@ public ResponseEntity<JsonNode> postProfileModel(
 		@PathVariable("model_id") String modelId,
 		@RequestParam("document_id") String documentId
 	) {
+
+		Provenance provenancePayload = new Provenance(ProvenanceRelationType.EXTRACTED_FROM, modelId, ProvenanceType.MODEL, documentId, ProvenanceType.PUBLICATION);
+		provenanceProxy.createProvenance(provenancePayload);
```

Comment:
Right now this and the other call to createProvenance is sort of shouting into the void - we should probably make sure that this succeeds and if not write something to the log (also if this call fails I believe it will throw an exception which will end wit this not returning properly.  You may want to do something like...

```

try{
     ResponseEntity<Provenance> r = provenanceProxy.createProvenance(provenancePayload);
     if (!r.getStatusCode().is2xxSuccessful())
        log.error("unable to create provenance");
} catch (Exception e) {
     log.error("unable to create provenance", e);
}

```

Reply:
Good point! I was thinking at the time I wouldn't want the provenance link to block the enrichment job from happening but thinking over it now, its probably better make sure the link happens.  Will make the change

Reply:
I think the way it is right now would end up blocking if there was any issues with the provenance call

Reply:
Fix this up! Too much javascript brain 😆